src/parser/jobs/smn/modules/DoTs.js

```diff
@@ -153,51 +173,61 @@ export default class DoTs extends Module {
 		return (statusUptime / fightDuration) * 100
 	}
 
-	output() {
+	_createTargetStatusTable(target) {
 		let totalBioClip = 0
 		let totalMiasmaClip = 0
 		return <Table collapsing unstackable style={{border: 'none'}}>
 			<Table.Body>
 				<Table.Row>
-					<Table.Cell style={{padding: '0 1em 0 0'}}>
+					<Table.Cell style={{padding: '0 1em 0 0', verticalAlign: 'top'}}>
 						<Table collapsing unstackable>
 							<Table.Header>
 								<Table.Row>
 									<Table.HeaderCell><ActionLink {...ACTIONS.MIASMA_III} /> <Trans id="smn.dots.applied">Applied</Trans></Table.HeaderCell>
 									<Table.HeaderCell><Trans id="smn.dots.clip">Clip</Trans></Table.HeaderCell>
 									<Table.HeaderCell><Trans id="smn.dots.total-clip">Total Clip</Trans></Table.HeaderCell>
+									<Table.HeaderCell><Trans id="smn.dots.source">Source</Trans></Table.HeaderCell>
 								</Table.Row>
 							</Table.Header>
 							<Table.Body>
-								{this._application[STATUSES.MIASMA_III.id].map(
+								{target[STATUSES.MIASMA_III.id].map(
 									(event) => {
 										totalMiasmaClip += event.clip
+										// Depending on latency of status application, the Bio III instant cast may have
+										// resolved before the previous Miasma III went off.  Assume that this the normal
+										// Miasma III -> Bio III cast pattern happened and adjust the id.
+										if (event.source === ACTIONS.BIO_III.id) { event.source = ACTIONS.MIASMA_III.id }
```

Comment:
Would it be better to do this check in the _pushApplication routine, rather than this output-related method having side effects?

Could the "assume they went Miasma -> Bio" assumption be mitigating by tracking lastMiasmaCast and lastBioCast separately, and just having TriD and Bane update both of those values?

Reply:
This was mostly a "simplest thing that worked" situation.  These are now split in the new commits.